ygot/render.go

```diff
@@ -143,7 +143,10 @@ func (g *gnmiPath) isPathElemPath() bool {
 	return g.pathElemPath != nil
 }
 
-// Copy returns a copy of the current gnmiPath.
+// Copy returns a shallow copy of the current gnmiPath.
+//
+// In particular, the pre-0.4.0 gNMI path is a deep copy, but the PathElem gNMI
```

Comment:
Should we make this a deeper copy? It'll have some overhead to do it with `proto.Clone`, but I presume that the shallow copy caused an issue somewhere?

Reply:
I'll add a flag to `GNMINotificationsConfig` for doing deep copying.

The reason `proto.Clone` is not used in relevant locations is because it would negatively impact the performance of `ToGNMINotifications`: https://github.com/openconfig/ygot/pull/609

Reply:
Makes sense. What was the trigger for this CL? If we don't care about deep-copying then no need to add a flag, but I assume someone edited something somewhere and it caused issues?

Reply:
Yes Stuart was writing something to test config/state and was converting the config -> state and that caused the issue.

Here we can either make shallow-copy the default behaviour or deep-copy the default behaviour.

Reply:
I'd prefer shallow-copy to be the default behaviour since someone might be relying on the performance characteristic. So I'll merge this and add the DeepCopy flag.

Reply:
Actually I think it's ok that we simply keep the comment. The user can just deepcopy the returned notification per the doc comment.